test/unified_malloc_framework/memoryProviderAPI.cpp

```diff
@@ -23,7 +23,8 @@ TEST_F(test, memoryProviderTrace) {
 
     size_t call_count = 0;
 
-    auto ret = umfMemoryProviderAlloc(tracingProvider.get(), 0, 0, nullptr);
+    void *ptr;
```

Comment:
Is this and `*ptr = NULL;` in `provider.c` needed?

Reply:
Yes, this is because the implementation handles NULL pointers differently now (so we need to special case it in the tracker for both alloc and free).